pyomeca/markers.py

```diff
@@ -1,180 +1,513 @@
+            ```
         """
-        s = m.shape
-        if s[1] % 3 != 0:
-            raise IndexError("Number of columns must be divisible by 3")
-        return Markers3d(np.reshape(m.T, (3, int(s[1] / 3), s[0]), "F"))
+        return Markers(getattr(np.random, distribution)(size=size, **kwargs).cumsum(-1))
 
     @classmethod
-    def from_trc(cls, filename):
-        return cls.from_csv(
+    def from_csv(
+        cls,
+        filename: Union[str, Path],
+        usecols: Optional[List[Union[str, int]]] = None,
+        header: Optional[int] = None,
+        first_row: int = 0,
+        first_column: Optional[Union[str, int]] = None,
+        time_column: Optional[Union[str, int]] = None,
+        last_column_to_remove: Optional[Union[str, int]] = None,
+        prefix_delimiter: Optional[str] = None,
+        suffix_delimiter: Optional[str] = None,
+        skiprows: Optional[List[int]] = None,
+        pandas_kwargs: Optional[dict] = None,
+        attrs: Optional[dict] = None,
+    ) -> xr.DataArray:
+        """
+        Markers DataArray from a csv file.
+
+        Arguments:
+            filename: Any valid string path
+            usecols: All elements must either be positional or strings that correspond to column names.
+                For example, a valid list-like usecols parameter would be [0, 1, 2] or ['foo', 'bar', 'baz'].
+            header: Row of the header (0-indexed)
+            first_row: First row of the data (0-indexed)
+            first_column: First column of the data (0-indexed)
+            time_column: Location of the time column. If None, indices are associated
+            last_column_to_remove: If for some reason the csv reads extra columns, how many should be ignored
+            prefix_delimiter: Delimiter that split each column name by its prefix (we keep only the column name)
+            suffix_delimiter: Delimiter that split each column name by its suffix (we keep only the column name)
+            skiprows: Line numbers to skip (0-indexed)
+            pandas_kwargs: Keyword arguments to be passed to `pandas.read_csv`
+            attrs: attrs to be passed to `xr.DataArray`. If attrs['rate'] is provided, compute the time accordingly
+
+        Returns:
+            Markers `xarray.DataArray` with the specified data and coordinates
+
+        !!! example
+            To read [this csv file](https://github.com/pyomeca/pyomeca/blob/master/tests/data/markers.csv),
+            type:
+
+            ```python
+            from pyomeca import Markers
+
+            data_path = "./tests/data/markers.csv"
+            markers = Markers.from_csv(data_path, header=2, first_row=5, first_column=2)
+            ```
+
+            If you know the channel names, you can retrieve only the ones you are interested in by specifying strings:
+
+            ```python
+            channels = ["Daphnee:ASISr", "Daphnee:ASISl", "Daphnee:PSISr"]
```

Comment:
That a tribute to Daphnee? :P

Reply:
Yes haha! It is the only `c3d` that I had access so... :laughing: